src/vm/object_model.rs

```diff
@@ -248,6 +248,9 @@ pub trait ObjectModel<VM: VMBinding> {
     /// Arguments:
     /// * `object`: The object to be dumped.
     fn dump_object(object: ObjectReference);
+
+    /// Return the object alignment setting of the vm
+    fn object_alignment() -> i32;
```

Comment:
OpenJDK has the vm flag ObjectAlignmentInBytes.

Reply:
> -XX:ObjectAlignmentInBytes=alignment
Sets the memory alignment of Java objects (in bytes). By default, the value is set to 8 bytes. The specified value should be a power of two, and must be within the range of 8 and 256 (inclusive). This option makes it possible to use compressed pointers with large Java heap sizes.
>
> The heap size limit in bytes is calculated as:
>
> 4GB * ObjectAlignmentInBytes

Does this mean all OpenJDK objects have the same alignment, specified by this option between [8, 256]? 

I don't think we can assume all the VMs have this kind of property that all the objects have the same alignment. For VMs that use different alignment values for different types, they can't provide an implementation for this method.

Reply:
Yes, all OpenJDK objects will have the same alignment. 

For those VMs, they can return the least common multiple of all the alignment.

Reply:
This still sounds confusing to me. What you are saying is this method will return an alignment value, but sometimes it is the same alignment for all the objects and sometimes it is the max alignment for all the objects. This does not clearly define what this method will return.

In this case, for internal use in mmtk core, why would I use this method rather than `MAX_ALIGNMENT`? It sounds like this method does not give me more information than the `MAX_ALIGNMENT`.

Reply:
The object alignment configuration is a VM specific thing. I did not know that VM might have different alignment for different types of objects when I added this api. Ideally, we should rely on ````align`: Required alignment for the object.``` but that is hardcoded as the size of a word in openjdk binding.

Reply:
> Ideally, we should rely on `align: Required alignment for the object.`

We can rely on `align` in `alloc()`. But once alloc is done, we do not know `align` for the object.

>but that is hardcoded as the size of a word in openjdk binding.

If the binding gives us a wrong `align` value for `alloc()`, that is the binding's bug, and it does not affect how mmtk core is implemented. It seems what you are saying is the OpenJDK binding basically ignores the `ObjectAlignmentInBytes` option and always use 1 word as align. If that is intended, that is fine. That means our OpenJDK binding does not support the `ObjectAlignmentInBytes` option.

Reply:
I check the ```MAX_ALIGNMENT``` in ```VMBinding``` and its value is ```8```. I am going to use it for now